app/src/main/java/com/github/se/eventradar/viewmodel/qrCode/QrCodeAnalyser.kt

```diff
@@ -19,7 +19,8 @@ class QrCodeAnalyser @Inject constructor() : ImageAnalysis.Analyzer {
   private val supportedImageFormats =
       listOf(ImageFormat.YUV_420_888, ImageFormat.YUV_422_888, ImageFormat.YUV_444_888)
 
-  var onDecoded: ((String?) -> Unit)? = null //
+  var onFriendDecoded: ((String?) -> Unit)? = null
+  var onTicketDecoded: ((String?) -> Unit)? = null
```

Comment:
Is it really necessary to have two variables here ? One for the friend case and one for ticket case ? I see that below you use an if/else to separate the cases, can't it be just a generic var `onQrDecoded` that you call at the end of the decoding ? Since you said in comment below that it has to be reset to null by the viewModel later

Reply:
Yes, I was just being over precautious ... changed it all now I am certain :)